src/platform/BUILD.gn

```diff
@@ -318,6 +318,8 @@ if (chip_device_platform != "none") {
       sources += [
         "GenericOTARequestorDriver.cpp",
         "GenericOTARequestorDriver.h",
+        "GenericOTARequestorStorage.cpp",
+        "GenericOTARequestorStorage.h",
```

Comment:
If this is added to OTA requestor, it means it's not possible for an implementer to provide their own impl of an `OTARequestorStorage` interface without paying/building for this one.

Suggest creating a new library in `examples/<somehwere>`, so that it's possible for different platforms to choose to use this example, or completely use something different when they initialize the OTARequestor and pass in an instance *they* provide of `OTARequestorStorage`

Reply:
I think we only have unit tests for the code in `src/` and I think despite the theoretical possibility to use a custom implemention, most applications will probably use the generic one, so it would be good to unit-test it. Maybe I should add more switches to enable the generic OTA code?

Reply:
It's fine if "most implementations use it", but the way the build target is organized, it's on the path to "all implementations must use it".

An alternative would be to split the OTA requestor feature outside of the main `libDeviceLayer`, and make a new static  lib that targets can link against, rather than building this in main libCHIP

Reply:
Ok, I'll the unit tests and will be able to do it.

Reply:
Looking at the existing classes like `GroupDataProviderImpl` they are actually part of `src/` probably to make them unit-tested. After second thought, IMHO it's important to have that persistence layer unit unit-tested so I kept that file where it is and added `chip_enable_ota_requestor_generic_storage` if someone doesn't want to compile that file.

Reply:
Yes, it's OK to steer away from a default with a build flag, As long as that works "for real" (i.e. it will actually still build properly if you provide external version), it's a good way to go to override SDK "defaults".